packages/cli/src/workflows/workflows.services.ts

```diff
@@ -175,12 +183,26 @@ export class WorkflowsService {
 		}
 
 		filter.id = In(sharedWorkflowIds);
-		return Db.collections.Workflow.find({
+
+		if (typeof filter.name === 'string' && filter.name !== '') {
+			filter.name = Like(`%${filter.name}%`);
+		}
+
+		const findManyOptions: FindManyOptions<WorkflowEntity> = {
 			select,
 			relations,
 			where: filter,
-			order: { updatedAt: 'DESC' },
-		});
+			order: { updatedAt: 'ASC' },
+		};
+
+		if (options?.take) {
+			const { skip, take } = WorkflowRepository.toPaginationOptions(options.take, options.skip);
+
+			findManyOptions.skip = skip;
+			findManyOptions.take = take;
+		}
+
+		return Db.collections.Workflow.findAndCount(findManyOptions);
```

Comment:
most of this code can be moved into the repository class. eventually no files outside the `src/database` folder should have any references to typeorm

Reply:
Agree but this should be a separate effort to refactor all relevant methods in this service, not just this one. Not addressing here to avoid bloating this PR.